VideoWeb/VideoWeb/ClientApp/src/app/services/hearing-layout.service.ts

```diff
@@ -0,0 +1,119 @@
+                this.eventsService
+                    .getHearingLayoutChanged()
+                    .pipe(
+                        takeUntil(
+                            this.conferenceService.currentConference$.pipe(filter(conference => conference?.id !== currentConferenceId))
+                        ),
+                        tap(layoutChanged => {
+                            if (layoutChanged.conferenceId === currentConferenceId) {
+                                return;
+                            }
+
+                            this.logger.debug(
+                                `${this.loggerPrefix} layout changed from ${layoutChanged.newHearingLayout} to ${layoutChanged.newHearingLayout} for conference: ${currentConferenceId}`
+                            );
+                        }),
+                        filter(layoutChanged => layoutChanged.conferenceId === currentConferenceId)
+                    )
+                    .subscribe(layoutChanged => {
+                        this.logger.info(
+                            `${this.loggerPrefix} layout changed from ${layoutChanged.newHearingLayout} to ${layoutChanged.newHearingLayout} in current conference: ${currentConferenceId}`
+                        );
+                        this.currentLayoutSubject.next(layoutChanged.newHearingLayout);
+                    });
+            });
+    }
+
+    getCurrentLayout(): Observable<HearingLayout> {
+        return this.conferenceService.currentConference$.pipe(
+            take(1),
+            map(conference => conference.id),
+            mergeMap(currentConferenceId => this.apiClient.getLayoutForHearing(currentConferenceId))
+        );
+    }
+
+    updateCurrentLayout(layout: HearingLayout) {
+        this.conferenceService.currentConference$
+            .pipe(
+                take(1),
+                map(conference => conference.id)
+            )
+            .subscribe(currentConferenceId => {
+                this.logger.info(
+                    `${this.loggerPrefix} updating current layout to ${layout} for current conference: ${currentConferenceId}`
+                );
+                this.eventsService.updateHearingLayout(currentConferenceId, layout);
+            });
+    }
+
+    get recommendedLayout$(): Observable<HearingLayout> {
+        return this.conferenceService.currentConference$.pipe(
+            distinctUntilChanged((x, y) => x?.id === y?.id),
+            map(conference => {
+                const numOfParticipantsIncJudge = conference.participants.length + conference.endpoints.length;
+
+                if (numOfParticipantsIncJudge >= 10) {
+                    return HearingLayout.TwoPlus21;
+                }
+
+                if (numOfParticipantsIncJudge >= 6 && numOfParticipantsIncJudge <= 9) {
+                    return HearingLayout.OnePlus7;
+                }
+
+                return HearingLayout.Dynamic;
```

Comment:
These numbers will now have to be maintained in 2 places if there is any changes. Would there be any value to having the recommended layout be determined on the backend? Although there's issues then if the participant count changes afterwards.

Reply:
I will make this a call to the backend and also reemit the value if the participant list changes